BuoyTest/src/components/cocktails/CocktailsList.js

```diff
@@ -0,0 +1,45 @@
+import React, { Component } from 'react';
+import { Stylesheet, View, FlatList, Text, Image, TouchableWithoutFeedback } from 'react-native';
+import styles from './styles';
+
+class CocktailsList extends Component {
+  state = {}
+  tableHeader = () => (
+    <View>
+    </View>
+  )
+
+  renderRows(object, indexRow) {
+    const cocktail = this.props.cocktails.find(cocktail => cocktail.idDrink === object.idDrink);
+    return (
+      <TouchableWithoutFeedback onPress={() => this.props.onItemPress(cocktail)} >
+        <View style={styles.cell}>
+          <Image
+            style={styles.photo}
+            source={{uri: cocktail.strDrinkThumb}}
+          />
+          <Text>{cocktail.strDrink}</Text>
+        </View>
+      </TouchableWithoutFeedback >
+    );
+  }
+
+  render() {
+    const { cocktails, refreshing, emptyMessage } = this.props;
+    return (
+      <View>
+        {this.tableHeader()}
+        <FlatList
+          refreshing={refreshing}
+          data={cocktails}
+          ItemSeparatorComponent={() => <View style={styles.separatorStyle} />}
+          keyExtractor={(item) => item.idDrink}
+          ListEmptyComponent={() => <Text style={styles.emptyListStyle}>{emptyMessage}</Text>}
+          renderItem={({ item, index }) => this.renderRows(item, index)}
```

Comment:
Here, you can just do
`renderItem={this.renderRows}`
FlatList will call your function with the same parameters you are defined

Reply:
The change did not work for me. The cells are not loaded if I use renderRows directly without parameters.

Reply:
Yes, but that is because your `renderRows` is not defined withe the same arguments than `renderItem`.
Pay attention to this: `renderItem` is being called with just one argument, an object that contains 2 keys, item and index. Your fn is declared with 2 arguments.

In order to make it work, refactor your function like this `const renderRow = ({item, index}) =>`

Reply:
Yes, that makes sense. I changed and it works. Thanks!